tests/unit/engines/test_libvirt_events.py

```diff
@@ -0,0 +1,97 @@
+    '''
+
+    def setup_loader_modules(self):
+        patcher = patch('salt.engines.libvirt_events.libvirt')
+        self.mock_libvirt = patcher.start()
+        self.mock_libvirt.getVersion.return_value = 2000000
+        self.addCleanup(patcher.stop)
+        self.addCleanup(delattr, self, 'mock_libvirt')
+        return {libvirt_events: {}}
+
+    def test_get_libvirt_enum_string(self):
+        '''
+        Make sure the libvirt enum value to string works reliably
+        '''
+        with patch('salt.engines.libvirt_events.libvirt') as mock_libvirt:
+            # Test case with a sub prefix
+            with patch.dict(mock_libvirt.__dict__, {
+                    'VIR_PREFIX_NONE': 0,
+                    'VIR_PREFIX_ONE': 1,
+                    'VIR_PREFIX_TWO': 2,
+                    'VIR_PREFIX_SUB_FOO': 0,
+                    'VIR_PREFIX_SUB_BAR': 1,
+                    'VIR_PREFIX_SUB_FOOBAR': 2}):
+                assert libvirt_events._get_libvirt_enum_string('VIR_PREFIX_', 2) == 'two'
+
+            # Test case with items including underscores
+            with patch.dict(mock_libvirt.__dict__, {
+                    'VIR_PREFIX_FOO': 0,
+                    'VIR_PREFIX_FOO_BAR': 1,
+                    'VIR_PREFIX_BAR_FOO': 2}):
+                assert libvirt_events._get_libvirt_enum_string('VIR_PREFIX_', 1) == 'foo bar'
+
+
+    def test_get_domain_event_detail(self):
+        '''
+        Test get_domain_event_detail function
+        '''
+        with patch('salt.engines.libvirt_events.libvirt') as mock_libvirt:
+            with patch.dict(mock_libvirt.__dict__, {
+                    'VIR_DOMAIN_EVENT_DEFINED': 0,
+                    'VIR_DOMAIN_EVENT_UNDEFINED': 1,
+                    'VIR_DOMAIN_EVENT_DEFINED_ADDED': 0,
+                    'VIR_DOMAIN_EVENT_DEFINED_UPDATED': 1}):
+                assert libvirt_events._get_domain_event_detail(1, 2) == ('undefined', 'unknown')
+                assert libvirt_events._get_domain_event_detail(0, 1) == ('defined', 'updated')
+                assert libvirt_events._get_domain_event_detail(4, 2) == ('unknown', 'unknown')
+
+    def test_event(self):
+        '''
+        Test that the libvirt_events engine actually registers events catch them and cleans
+        before leaving the place.
+        '''
+        with patch('salt.engines.libvirt_events.libvirt') as mock_libvirt:
+            mock_libvirt.virEventRunDefaultImpl.return_value = -1
+            with patch.dict(mock_libvirt.__dict__, {
+                    'VIR_DOMAIN_EVENT_ID_LIFECYCLE': 0,
+                    'VIR_DOMAIN_EVENT_DEFINED': 0,
+                    'VIR_DOMAIN_EVENT_DEFINED_ADDED': 0,
+                    'VIR_NETWORK_EVENT_ID_LIFECYCLE': 0,
+                    'VIR_NETWORK_EVENT_STOPPED': 3}):
+                uri = 'test:///'
+                prefix = 'test/prefix'
+
+                libvirt_events.start(uri, prefix)
+
+                assert call.call_list() != []
```

Comment:
I am not sure you're using this correctly: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.call